src/psij/launchers/script_based_launcher.py

```diff
@@ -98,26 +98,29 @@ def _ensure_launcher_deployed(self) -> None:
                 return
 
             self._deploy_file(Path(__file__).parent / 'scripts' / 'launcher_lib.sh')
-            self._deploy_file(self._script_path)
+            self._deployed_script_path = self._deploy_file(self._script_path)
 
-    def _deploy_file(self, path: Path) -> None:
+    def _deploy_file(self, path: Path) -> Path:
         dst_dir = self.config.work_directory
         dst_dir.mkdir(parents=True, exist_ok=True)
         dst_path = dst_dir / path.name
         if dst_path.exists():
-            return
+            return dst_path
         tmp_prefix = secrets.token_hex() + '_'
         tmp_path = dst_dir / (tmp_prefix + path.name)
         shutil.copy(path, tmp_path)
         try:
             # this appears to use os.rename, although I'd wish pathlib docs would state this
             # explicitly, since the docs for Path.rename mention nothing of the exceptions thrown
             tmp_path.rename(dst_path)
+            return dst_path
         except FileExistsError:
             # thrown in Windows if the path already exists, which is fine if the destination is a
             # file; we were recing another process
```

Comment:
recing -> racing?

Reply:
pfft "racing". "recing" is all the rage now.